ee/secureenclavesigner/secureenclavesigner_darwin.go

```diff
@@ -0,0 +1,253 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/kolide/launcher/ee/consoleuser"
+)
+
+const (
+	CreateKeyCmd     = "create-key"
+	PublicEccDataKey = "publicEccData"
+)
+
+type opt func(*secureEnclaveSigner)
+
+type secureEnclaveSigner struct {
+	uidPubKeyMap         map[string]*ecdsa.PublicKey
+	pathToLauncherBinary string
+	store                types.GetterSetterDeleter
+	slogger              *slog.Logger
+	mux                  *sync.Mutex
+}
+
+func New(slogger *slog.Logger, store types.GetterSetterDeleter, opts ...opt) (*secureEnclaveSigner, error) {
```

Comment:
I think it would be nice throw some traces in here:

```
ctx, span := traces.StartSpan(<parent ctx>)
defer span.End()
```

(if it's not too much of a pain to pass in a parent ctx too) -- I'm suggesting it because I think this happens on launcher startup, so it would be helpful to get this into our startup span so we can see timing details.

Reply:
Added some tracing, it was pretty easy to pass ctx